contrib/natss/config/broker/natss.yaml

```diff
@@ -62,8 +62,6 @@ spec:
       app: nats-streaming
   template:
     metadata:
-      annotations:
```

Comment:
What happens if we use the new dispatcher with the old NATSS (the one with a sidecar)?

Reply:
This Natss installation creates a StatefulSet and a Service (nats-streaming) with selector. The fallback default url in dispatcher is fallbackDefaultNatssURLTmpl = "nats://nats-streaming.natss.svc.<cluster-domain>:4222". 
Hence I am not sure why the sidecard was present in first place. I haven't tested the exact scenario, but from what I see, it should work.
Moreover, the dispatcher still has a sidecar. Except for new hostName resolution change the dispatcher is still the same when it comes to istio functionality.

Do you have a specific concern or a specific edge case?

Reply:
My specific concern is that both of the following work.

1. My 0.5 NATSS installation, which has the sidecar, continues to work with the new 0.6 controller and dispatcher.
1. My 0.6 NATSS installation, which doesn't have the sidecar, works with the new 0.6 controller and dispatcher.

This comment was meant to address scenario 1 above, as I believe scenario 2 will be covered by people following the installation instructions (and hopefully e2e tests soon!).

I'm iffy about what happens when one side of a connection has the sidecar and the other doesn't (dispatcher -> NATSS in this case). I just want to verify that it will work.

Reply:
Let me test that.